### PR TITLE
 corefile: also expose page_offset

### PR DESCRIPTION
**Pull request by Mic92**
# Pwntools Pull Request

Adds page_offset to corefiles. This is useful to data from the original files.

This pull request depends on: https://github.com/Gallopsled/pwntools/pull/1167

## Testing

tested via testsuite and:

```console
$ gdb ls -batch -ex 'break _start' -ex run -ex 'generate-core-file core'
```

```python
from pwn import Corefile
coredump = Corefile('./core')
print(coredump.mappings)
```

### CODE REVIEW

**Comment by zachriggle** [comment]
Can you rebase this on `dev` or revert aac086d in this branch?

**Comment by Mic92** [comment]
Rebased...

**Comment by zachriggle** [comment]
I made some quick changes to preserve the original ordering of the constructor arguments -- I'll merge this if/when CI passes 😄

File: pwnlib/elf/corefile.py [hunk 1]
```diff
@@ -184,12 +187,13 @@ def __str__(self):
         return '%x-%x %s %x %s' % (self.start,self.stop,self.permstr,self.size,self.name)
 
     def __repr__(self):
-        return '%s(%r, start=%#x, stop=%#x, size=%#x, flags=%#x)' \
+        return '%s(%r, start=%#x, stop=%#x, size=%#x, page_offset=%#x, flags=%#x)' \
```
Comment (Mic92) [inline pwnlib/elf/corefile.py]: What about this one?

File: pwnlib/elf/corefile.py [hunk 2]
```diff
@@ -295,21 +299,21 @@ class Corefile(ELF):
...
-         Mapping('/lib/i386-linux-gnu/libc-2.19.so', start=0xf76d2000, stop=0xf76d4000, size=0x2000, flags=0x4),
-         Mapping('/lib/i386-linux-gnu/libc-2.19.so', start=0xf76d4000, stop=0xf76d5000, size=0x1000, flags=0x6),
-         Mapping(None, start=0xf76d5000, stop=0xf76d8000, size=0x3000, flags=0x6),
-         Mapping(None, start=0xf76ef000, stop=0xf76f1000, size=0x2000, flags=0x6),
-         Mapping('[vdso]', start=0xf76f1000, stop=0xf76f2000, size=0x1000, flags=0x5),
-         Mapping('/lib/i386-linux-gnu/ld-2.19.so', start=0xf76f2000, stop=0xf7712000, size=0x20000, flags=0x5),
-         Mapping('/lib/i386-linux-gnu/ld-2.19.so', start=0xf7712000, stop=0xf7713000, size=0x1000, flags=0x4),
-         Mapping('/lib/i386-linux-gnu/ld-2.19.so', start=0xf7713000, stop=0xf7714000, size=0x1000, flags=0x6),
-         Mapping('[stack]', start=0xfff3e000, stop=0xfff61000, size=0x23000, flags=0x6)]
+        [Mapping('/home/user/pwntools/crash', start=0x8048000, stop=0x8049000, size=0x1000, page_offset=0x0, flags=0x5),
```
Comment (Mic92) [inline pwnlib/elf/corefile.py]: and here?

**Comment by zachriggle** [comment]
It shouldn’t matter since:

1) The doctest is still correct (repr() returns that string)
2) That repr() is still valid since it uses **kwargs instead of ordered arguments

**Comment by Mic92** [comment]
I just toughed it is more consistent.

**Comment by zachriggle** [comment]
Definitely -- feel free to fix it if you'd like, I just didn't want to make a large edit in the browser (and still ended up needing to make 3 edits 😅)

**Comment by Mic92** [comment]
Done. The current failure is just a temporary connection problem to ftp.ubuntu.org: https://travis-ci.org/Gallopsled/pwntools/builds/391975461#L1456

**Comment by zachriggle** [comment]
Restarted the CI build after fixing #1172

**Pull request merged**